src/app/_services/api/accounts/profile-details.service.ts

```diff
@@ -1,39 +1,32 @@
 import {Injectable} from '@angular/core';
-import {environment} from '@environments/environment';
 import {HttpClient} from '@angular/common/http';
-import {map} from 'rxjs/operators';
+import {catchError} from 'rxjs/operators';
 import {User} from '@app/_models';
 import {Observable} from 'rxjs';
+import {ApiService} from "@app/_services/api.service";
 
 @Injectable({
     providedIn: 'root'
 })
 export class ProfileDetailsService {
-    private profileDetailsUrl = new URL('/api/update-profile/', environment.apiBaseUrl).toJSON();
 
-    constructor(private http: HttpClient) {
+    constructor(private http: HttpClient, private apiService: ApiService) {
     }
 
     putProfileDetails(input: {
         first_name: string,
         last_name: string,
         email: string
     }, id: number): Observable<string> {
```

Comment:
Check the response type to make sure it's actually a string. If it's not modify all the occurrences.